### ISSUE TITLE
an error in eval.py

### ISSUE THREAD

**Issue by kapness** [toandaominh1997/EfficientDet.Pytorch#40]
get_augmentation func has no phase named 'valid', so eval.py throws a runtimeerror.

**Comment by huohuotm** [comment]
hi， 
if phase not in ['train', 'test'], it should run this following lines


`list_transforms = []
list_transforms.extend([
                        albu.Normalize(mean=(0.485, 0.456, 0.406),std=(0.229, 0.224, 0.225), p=1),
                        ToTensor()
                        ])
albu.Compose(list_transforms, 
                        bbox_params=albu.BboxParams(format='pascal_voc', 
                                                    min_area=min_area,
                                                    min_visibility=min_visibility, 
                                                    label_fields=['category_id']))
`

**Comment by kapness** [comment]
> hi，
> if phase not in ['train', 'test'], it should run this following lines
> list_transforms = []
> list_transforms.extend([ albu.Normalize(mean=(0.485, 0.456, 0.406),std=(0.229, 0.224, 0.225), p=1), ToTensor() ])
> albu.Compose(list_transforms,  bbox_params=albu.BboxParams(format='pascal_voc',  min_area=min_area, min_visibility=min_visibility,  label_fields=['category_id']))

yes, but it throws a runtimeerror... I think maybe I should resize the images and then feed them to the net, just like the 'test' mode does.

**Comment by Imposingapple** [comment]
I meet the same question, and I have printed out all the feature map sizes. The problem I think is miss of resizing the input image. I also don't know the solution, and I change the 'valid' to 'train' and 'test' and find that the mAp is very low(13% for train and 0.6% for test).  Is there any solution to the problem we have met?

**Comment by kapness** [comment]
exactly the same as me



---Original---

**Comment by pangpanghuahuamaomi** [comment]
> I meet the same question, and I have printed out all the feature map sizes. The problem I think is miss of resizing the input image. I also don't know the solution, and I change the 'valid' to 'train' and 'test' and find that the mAp is very low(13% for train and 0.6% for test).  Is there any solution to the problem we have met?

HIi, I have met the same problem as yours, now,  i run eval.py, map = 0.05, and some classes map is 0.
have you solved this problem? thanks.

**Comment by y78h11b09** [comment]
I also meet the same question, in validating , i changed the valid to 'test', and so, i get the mAp is 0.197 on VOC2012 Val.  
The model was trained 187 epoches on Voc0712
 last   cls_loss        : 0.1580013930797577
 last    reg_loss       : 0.22156712412834167
  last  mean_loss         :  0.5689369416236878
Can anybody help me ？ thanks
--

**Comment by y78h11b09** [comment]
I also meet the same question, in validating , i changed the valid to 'test', and so, i get the mAp is 0.197 on VOC2012 Val.
The model was trained 187 epoches on Voc0712
last cls_loss : 0.1580013930797577
last reg_loss : 0.22156712412834167
last mean_loss : 0.5689369416236878
Can anybody help me ？ thanks

**Comment by y78h11b09** [comment]
> I meet the same question, and I have printed out all the feature map sizes. The problem I think is miss of resizing the input image. I also don't know the solution, and I change the 'valid' to 'train' and 'test' and find that the mAp is very low(13% for train and 0.6% for test). Is there any solution to the problem we have met?

hi, have you solved the  question? thanks

**Comment by y78h11b09** [comment]
> > I meet the same question, and I have printed out all the feature map sizes. The problem I think is miss of resizing the input image. I also don't know the solution, and I change the 'valid' to 'train' and 'test' and find that the mAp is very low(13% for train and 0.6% for test).  Is there any solution to the problem we have met?
> 
> HIi, I have met the same problem as yours, now, i run eval.py, map = 0.05, and some classes map is 0.
> have you solved this problem? thanks.

hi, have you solved the question? thanks
